packages/cli/src/zosfiles/ZosFilesBase.handler.ts

```diff
@@ -39,33 +39,39 @@ export abstract class ZosFilesBaseHandler implements ICommandHandler {
      *
      * @returns {Promise<void>}
      */
-    public async process(commandParameters: IHandlerParameters) {
-        const sessCfg: ISession = ZosmfSession.createSessCfgFromArgs(
-            commandParameters.arguments
-        );
-        const sessCfgWithCreds = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(
-            sessCfg, commandParameters.arguments, {parms: commandParameters}
-        );
+    public async process(commandParameters: IHandlerParameters): Promise<void> {
+        try {
+            const sessCfg: ISession = ZosmfSession.createSessCfgFromArgs(
+                commandParameters.arguments
+            );
+            const sessCfgWithCreds = await ConnectionPropsForSessCfg.addPropsOrPrompt<ISession>(
+                sessCfg, commandParameters.arguments, { parms: commandParameters }
+            );
+            const session = new Session(sessCfgWithCreds);
 
-        const session = new Session(sessCfgWithCreds);
-        const response = await this.processWithSession(commandParameters, session);
-
-        commandParameters.response.progress.endBar(); // end any progress bars
-        // Print out the response
-        if (response.commandResponse) {
-            commandParameters.response.console.log(response.commandResponse);
-        }
-
-        // Return as an object when using --response-format-json
-        commandParameters.response.data.setObj(response);
-
-        // Ensure error gets thrown if request was unsuccessful.
-        // Sometimes it is useful to delay throwing an error until the end of the handler is
-        // reached, for example the upload API needs to return an API response even when it fails.
-        if (!response.success && response.commandResponse) {
-            throw new ImperativeError({
-                msg: response.errorMessage || response.commandResponse
-            });
+            const response = await this.processWithSession(commandParameters, session);
+            if (!response.success && response.commandResponse) {
+                throw new ImperativeError({
+                    msg: response.errorMessage || response.commandResponse
+                });
+            }
+            if (response.commandResponse) {
+                commandParameters.response.console.log(response.commandResponse);
+            }
+            commandParameters.response.data.setObj(response);
+            commandParameters.response.progress.endBar();
```

Comment:
The integration tests are still failing because the order of these statements has been switched around. Restoring the original order should fix them 😋 
1. `commandParameters.response.progress.endBar();`
2. `if (response.commandResponse) ...`
3. `commandParameters.response.data.setObj(response);`
4. `if (!response.success && response.commandResponse) ...`